server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -213,7 +221,8 @@ export class MetadataExtractionProcessor {
         }
       }
 
-      const exifData = await exiftool.read<ImmichTags>(asset.originalPath).catch((error: any) => {
+      const sidecarPath = asset.sidecarPath || (fs.existsSync(`${asset.originalPath}.xmp`) ? `${asset.originalPath}.xmp` : null)
```

Comment:
Same logic again here again. Maybe we should make a private method `getSidecarPath(asset: AssetEntity)` that handles this logic`